File: app/src/main/java/com/rye/catcher/base/mvp/BasePresenter.java
package com.rye.catcher.base.mvp;

/**
 * Created by 18041at 2019/5/26
 * Function:
 */
public abstract class BasePresenter<T extends BaseView> {
   public T mView;
   public void attach(T mView){
       this.mView=mView;
   }
   public void  detach(){
       mView=null;
   }
}

File: base/src/main/java/com/rye/base/mvp/BasePresenter.kt
package com.rye.base.mvp

import java.lang.ref.WeakReference

/**
 * Create by rye
 * at 2020-08-06
 * @description:
 */
open class BasePresenter<V : BaseView> : BaseContract.IPresenter<V> {
    var mView: WeakReference<V>? = null
    override fun attachView(v: V) {
        mView = WeakReference(v)
    }

    override fun detachView() {
        mView = null
    }
}